CustomerSettingsActivity.java

```diff
@@ -0,0 +1,185 @@
+
+    private String userID;
+    private String mName;
+    private String mPhone;
+    private String mProfileImageUrl;
+
+    private Uri resultUri;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_customer_settings);
+
+        mNameField = (EditText) findViewById(R.id.name);
+        mPhoneField = (EditText) findViewById(R.id.phone);
+
+        mProfileImage = (ImageView) findViewById(R.id.profileImage);
+
+        mBack = (Button) findViewById(R.id.back);
+        mConfirm = (Button) findViewById(R.id.confirm);
+
+        mAuth = FirebaseAuth.getInstance();
+        userID = mAuth.getCurrentUser().getUid();
+        mCustomerDatabase = FirebaseDatabase.getInstance().getReference().child("Users").child("Customers").child(userID);
+
+        getUserInfo();
+
+        mProfileImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(Intent.ACTION_PICK);
+                intent.setType("image/*");
+                startActivityForResult(intent, 1);
+            }
+        });
+
+        mConfirm.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                saveUserInformation();
+            }
+        });
+
+        mBack.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+                return;
+            }
+        });
+    }
+    private void getUserInfo(){
+        mCustomerDatabase.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                if(dataSnapshot.exists() && dataSnapshot.getChildrenCount()>0){
+                    Map<String, Object> map = (Map<String, Object>) dataSnapshot.getValue();
+                    if(map.get("name")!=null){
+                        mName = map.get("name").toString();
+                        mNameField.setText(mName);
+                    }
+                    if(map.get("phone")!=null){
+                        mPhone = map.get("phone").toString();
+                        mPhoneField.setText(mPhone);
+                    }
+                    if(map.get("profileImageUrl")!=null){
+                        mProfileImageUrl = map.get("profileImageUrl").toString();
+                        Glide.with(getApplication()).load(mProfileImageUrl).into(mProfileImage);
+                    }
+                }
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        });
+    }
+
+
+
+    private void saveUserInformation() {
+        mName = mNameField.getText().toString();
+        mPhone = mPhoneField.getText().toString();
+
+        Map userInfo = new HashMap();

```

Comment:
Please add type definitions: `Map<keyType, ValType> userInfo = new HashMap<>();`

Reply:
Also, if you do not need a mutable map, you can just use:
```
... userInfo = Map.of(
   "name", mName,
   "phone", mPhone
);
```